testing-modules/testing-libraries-2/src/test/java/com/baeldung/environmentvariablesfortest/EnvironmentVariablesSystemStubsTestNGUnitTest.java

```diff
@@ -0,0 +1,27 @@
+package com.baeldung.environmentvariablesfortest;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
+import uk.org.webcompere.systemstubs.testng.SystemStub;
+import uk.org.webcompere.systemstubs.testng.SystemStubsListener;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Listeners(SystemStubsListener.class)
+public class EnvironmentVariablesSystemStubsTestNGUnitTest {
+    @SystemStub
+    private EnvironmentVariables setEnvironment;
+
+    @BeforeClass
+    public void beforeAll() {
+        setEnvironment.set("testng", "has environment variables");
+    }
+
+    @Test
+    public void givenEnvironmentVariableWasSet_thenItCanBeRead() {
+        assertThat(System.getenv("testng"))
+          .isEqualTo("has environment variables");
```

Comment:
I would usually expect to see single element assertions on a single line, even when using AssertJ - but I guess this is a matter of style so if you feel strongly about it, feel free to leave it